llvm/tools/llvm-readobj/llvm-readobj.cpp

```diff
@@ -360,8 +360,8 @@ createDumper(const ObjectFile &Obj, ScopedPrinter &Writer) {
 }
 
 /// Dumps the specified object file.
-static void dumpObject(ObjectFile &Obj, ScopedPrinter &Writer,
-                       const Archive *A = nullptr) {
+void ObjDumper::dumpObject(ObjectFile &Obj, ScopedPrinter &Writer,
```

Comment:
I don't like this: you're polluting the generic interface with parameters that are specific to one file format. It's okay to check `Obj.isCOFF()` in this function, and I'm also okay with it becoming an `ObjDumper` member (if moved to ObjDumper.cpp), but the hybrid object check really needs to be within the COFF-specific code.

One alternative approach might be to make the second `dumpObject` call, then call `printCOFFHybridObject` after that call. A variation of this might even be to introduce a new high-level function `dumpCOFFHybridObject`, much like the existing `dumpCOFFImportFile` etc that does this. (NB: I haven't given this a thorough thinking through, so this might not quite work, but hopefully it makes the principle clear).

Reply:
Thanks for the feedback. I tried moving `dumpObject` to `ObjDumper.cpp`, but that would require making a number of `opts` variables non-static and became a bit messy. Instead, I kept is static and changed it to be more similar to `dumpMachOUniversalBinary`: I used a separate branch in `dumpInput` for COFF files, where I call `dumpObject` twice in case for hybrid objects. That made the change quite simple.

Also, following the suggestion from #102071, I changed tests to use `--match-full-lines` and `--strict-whitespace` to make sure that indention works as expected.